### PR TITLE
 fix: solution for checkbox

### PR DESCRIPTION
**Pull request by libergod**
Solution for Issue #2 

Hi, 

Identified and checking. Will be opening an PR quick.

1) Problem with check boxes, wich returned the error:
PROBLEM:
ERROR: SQLSTATE[HY000]: General error: 1366 Incorrect integer value: '' for column 'account' at row 1

CAUSE:
When you have checkboxes, for some reason, if you didn't flag then, it will never come in the request. So, for some people works putting one more input checkbox hidden with default value 0, for me it didn't work, so i've made an logic in the Controller.
image

SOLUTION:
I know this isn't the best solution, but works. Later we can get this better. The main solution is validate if the postVars are being set. If not, we put a 0 on then.
image

2) Missing status feedback on submit

Currently working on it.

### CODE REVIEW

**Comment by gpedro** [comment]
can you try use `intval` instead `filter_var`? intval return 0 if isn't a int properly

**Comment by libergod** [comment]
> can you try use `intval` instead `filter_var`? intval return 0 if isn't a int properly

The problem isn't about the function itself. It's the missing validation of Post variables being sent or not to the script.

Don't know why POST function don't send checkboxes values if they aren't checked, but this were the path i'd take.

File: app/Controller/Pages/Account/CharacterEdit.php [hunk 1]
```diff
@@ -60,7 +81,8 @@ public static function updateCharacter($request, $name)
 
                 $check_exists_display = EntityPlayer::getDisplay('player_id = "'.$select_player->id.'"')->fetchObject();
                 if (empty($check_exists_display)) {
-                    EntityPlayer::insertDisplay($arrayCharacterDisplay);
+						EntityPlayer::insertDisplay($arrayCharacterDisplay);
+					return self::viewCharacterEdit($request, $name, 'Updated successfully.');
```
Comment (dudantas) [inline app/Controller/Pages/Account/CharacterEdit.php:85]:
```suggestion
                    EntityPlayer::insertDisplay($arrayCharacterDisplay);
                    return self::viewCharacterEdit($request, $name, 'Updated successfully.');
```

File: app/Controller/Pages/Account/CharacterEdit.php [hunk 2]
```diff
@@ -71,6 +93,8 @@ public static function updateCharacter($request, $name)
                         'bonus' => $filter_bonus,
                         'comment' => $filter_comment,
                     ]);
+					
+					return self::viewCharacterEdit($request, $name, 'Updated successfully.');
```
Comment (dudantas) [inline app/Controller/Pages/Account/CharacterEdit.php:97]:
```suggestion

                    return self::viewCharacterEdit($request, $name, 'Updated successfully.');
```

File: app/Controller/Pages/Account/CharacterEdit.php [hunk 3]
```diff
@@ -143,12 +167,13 @@ public static function getAchievementsPlayer($request, $name)
...
     }
 
-    public static function viewCharacterEdit($request, $name)
+    public static function viewCharacterEdit($request, $name, $status = null)
     {
         $content = View::render('pages/account/characteredit', [
             'player' => self::getCharacterEdit($request, $name),
             'achievements' => self::getAchievementsPlayer($request, $name),
             'total_secretachievements' => (int)EntityAchievements::getAchievements('secret = "1"', null, null, 'COUNT(*) as qtd')->fetchObject()->qtd,
+			'status' => $status,
```
Comment (dudantas) [inline app/Controller/Pages/Account/CharacterEdit.php:176]:
```suggestion
            'status' => $status,
```

File: app/Controller/Pages/Account/CharacterEdit.php [hunk 4]
```diff
@@ -33,12 +33,33 @@ public static function updateCharacter($request, $name)
             }
 
             $filter_account = filter_var($postVars['hide_account'], FILTER_SANITIZE_NUMBER_INT);
+			if(empty($filter_account)){
+				$filter_account = 0;
+			}
```
Comment (dudantas) [inline app/Controller/Pages/Account/CharacterEdit.php:38]:
```suggestion
            if(empty($filter_account)){
                $filter_account = 0;
            }
```

File: app/Controller/Pages/Account/CharacterEdit.php [hunk 5]
```diff
@@ -33,12 +33,33 @@ public static function updateCharacter($request, $name)
...
             $filter_outfit = filter_var($postVars['hide_outfit'], FILTER_SANITIZE_NUMBER_INT);
+			if(empty($filter_outfit)){
+				$filter_outfit = 0;
+			}
```
Comment (dudantas) [inline app/Controller/Pages/Account/CharacterEdit.php:42]:
```suggestion
            if(empty($filter_outfit)){
                $filter_outfit = 0;
            }
```

File: app/Controller/Pages/Account/CharacterEdit.php [hunk 6]
```diff
@@ -33,12 +33,33 @@ public static function updateCharacter($request, $name)
...
             $filter_inventory = filter_var($postVars['hide_inventory'], FILTER_SANITIZE_NUMBER_INT);
+			if(empty($filter_inventory)){
+				$filter_inventory = 0;
+			}
```
Comment (dudantas) [inline app/Controller/Pages/Account/CharacterEdit.php:46]:
```suggestion
            if(empty($filter_inventory)){
                $filter_inventory = 0;
            }
```

File: app/Controller/Pages/Account/CharacterEdit.php [hunk 7]
```diff
@@ -33,12 +33,33 @@ public static function updateCharacter($request, $name)
...
             $filter_health_mana = filter_var($postVars['hide_healthmana'], FILTER_SANITIZE_NUMBER_INT);
+			if(empty($filter_health_mana)){
+				$filter_health_mana = 0;
+			}
```
Comment (dudantas) [inline app/Controller/Pages/Account/CharacterEdit.php:50]:
```suggestion
            if(empty($filter_health_mana)){
                $filter_health_mana = 0;
            }
```

File: app/Controller/Pages/Account/CharacterEdit.php [hunk 8]
```diff
@@ -33,12 +33,33 @@ public static function updateCharacter($request, $name)
...
             $filter_skills = filter_var($postVars['hide_skills'], FILTER_SANITIZE_NUMBER_INT);
+			if(empty($filter_skills)){
+				$filter_skills = 0;
+			}
```
Comment (dudantas) [inline app/Controller/Pages/Account/CharacterEdit.php:54]:
```suggestion
            if(empty($filter_skills)){
                $filter_skills = 0;
            }
```

File: app/Controller/Pages/Account/CharacterEdit.php [hunk 9]
```diff
@@ -33,12 +33,33 @@ public static function updateCharacter($request, $name)
...
             $filter_bonus = filter_var($postVars['hide_bonus'], FILTER_SANITIZE_NUMBER_INT);
+			if(empty($filter_bonus)){
+				$filter_bonus = 0;
+			}
```
Comment (dudantas) [inline app/Controller/Pages/Account/CharacterEdit.php:58]:
```suggestion
            if(empty($filter_bonus)){
                $filter_bonus = 0;
            }
```

File: app/Controller/Pages/Account/CharacterEdit.php [hunk 10]
```diff
@@ -33,12 +33,33 @@ public static function updateCharacter($request, $name)
...
             $filter_comment = filter_var($postVars['comment'], FILTER_SANITIZE_SPECIAL_CHARS);
+			if(empty($filter_comment)){
+				$filter_comment = "";
+			}
```
Comment (dudantas) [inline app/Controller/Pages/Account/CharacterEdit.php:62]:
```suggestion
            if(empty($filter_comment)){
                $filter_comment = "";
            }
```

**Review by dudantas — APPROVED** [review]

**Pull request merged**